### DISCUSSION TITLE
Suppressing Warning Messages

### DISCUSSION THREAD

**Discussion by joniles** [ikvm-revived/ikvm-sdk-maven#73] [Q&A]
I look after a package publish on NuGet which makes use of `MavenReference`, something like this:

```xml
  <ItemGroup>
    <PackageReference Include="IKVM.Maven.Sdk" Version="1.8.2" />
    <MavenReference Include="net.sf.mpxj:mpxj" Version="13.7.0" />
  </ItemGroup>
```

I'm wondering if it is possible to supply some configuration to `MavenReference` to silence the IKVM warning messages end users see the first time they build their project after including my package? The messages are useful to me when things go wrong as a starting point for investigation, but under normal circumstances for end users of my package they are potentially confusing, or noise when emitted as part of a larger build process.

Any advice on the configuration options available for these messages gratefully received!

**Comment by wasabii** [comment]
Hadn't considered this.

**Comment by wasabii** [comment]
So, the vast majority of errors/warnings are class format errors come from multirelease JARs. We should fix that on our end some day. Pretty sure there is a bug in ikvm for it already.

The others are class format errors. Obviously getting IKVM supporting later JRE releases would fix that.

A few are class not found issues. I'd be curious if these are actual bugs in the Maven dep tree. Missing things? Maybe.

Not sure what approach to take here. I understand the user of YOUR PACKAGE don't want to see them. But I can't think of any coherent approach to distinguish you from everybody else.

**Reply by joniles** [reply]
The `MavenReference` configuration is part of my package. If in my package I could say something along the lines of:

```xml
  <ItemGroup>
    <PackageReference Include="IKVM.Maven.Sdk" Version="1.8.2" />
    <MavenReference Include="net.sf.mpxj:mpxj" Version="13.7.0" BuildQuietly="True"/>
  </ItemGroup>
```

That would allow me to request that warnings are not shown, but would only affect the build of that particular JAR (and its dependencies) when my package is included.

**Reply by wasabii** [reply]
Yeah but none of that information is preserved. It's all fed through Maven. Comes out the other side as an artifact tree.
